lib/bootloader_setup.pm

```diff
@@ -1040,37 +1040,95 @@ sub change_grub_config {
 
 =head2 add_grub_cmdline_settings
 
-    add_grub_cmdline_settings($add [, $update_grub ]);
+    add_grub_cmdline_settings($add [, $update_grub , $search]);
 
 Add C<$add> into /etc/default/grub, using sed.
 C<$update_grub> if set, regenerate /boot/grub2/grub.cfg with grub2-mkconfig and upload configuration.
+C<$search> if set, bypass default grub cmdline variable.
 =cut
 sub add_grub_cmdline_settings {
+    my (%args) = compat_args(
+        {
+            add         => '',
+            update_grub => 0,
+            search      => get_cmdline_var();
+        }, ['add', 'update_grub', 'search'], @_);
+
+    my $add         = %args{add};
+    my $update_grub = %args{update_grub};
+    my $search      = %args{search};
```

Comment:
You need to use `$` (scalar vs list context): `$args{priority}`

Reply:
Also travis told you that :)
https://travis-ci.org/os-autoinst/os-autoinst-distri-opensuse/jobs/581481735
`Global symbol "%args" requires explicit package name at lib/bootloader_setup.pm line 1057.`

It's good to 1) do run checks locally with `make tidy` 2) do code verification (safes time for reviewer).

Reply:
sorry, I mark a WIP for avoiding time waste. When It is ready, I will ping you.